modules/integration/src/test/scala/scala/cli/integration/BspTestDefinitions.scala

```diff
@@ -1446,6 +1452,17 @@ abstract class BspTestDefinitions(val scalaVersionOpt: Option[String])
     }
   }
 
+  test("source jars handled correctly from the command line") {
+    testSourceJars(passSourceJarFromCommandLine = true)
+  }
+
+  test("source jars handled correctly from a using directive") {
```

Comment:
It seems to me that this test could be a unit test to verify if the value for sourceJar is properly set in `BuildOptions`. The parsing of using directives can be tested in a unit test. What do you think?

Reply:
the source jar passed via directive has to be packaged first, so I don't think a unit test is suitable.
additionally, I think it makes sense to double-check we pass sources correctly in BSP `buildTarget/dependencySources`
So I'd rather keep it as an integration test

Reply:
For the unit test, we can create a dummy source JAR, which can be a simple text file with the .jar extension.

I think the integration test can be moved to a unit test:

- `"source jars handled correctly from a using directive"`
- `"source jars handled correctly from the command line smartly assuming a *-sources.jar is a source jar"`

Reply:
Done, I converted those 2 ITs to unit tests in the last commit